app/routers/applications.py

```diff
@@ -239,20 +239,22 @@ async def verify_data_field(
     with transaction_session(session):
         # Update a specific field in the application's `secop_data_verification` attribute.
         payload_dict = {key: value for key, value in payload.dict().items() if value is not None}
-        key, value = next(iter(payload_dict.items()), (None, None))
-        verified_data = application.secop_data_verification.copy()
-        verified_data[key] = value
-        application.secop_data_verification = verified_data.copy()
-
-        models.ApplicationAction.create(
-            session,
-            type=models.ApplicationActionType.DATA_VALIDATION_UPDATE,
-            data=jsonable_encoder(payload, exclude_unset=True),
-            application_id=application.id,
-            user_id=user.id,
-        )
+        try:
+            key, value = next(iter(payload_dict.items()), (None, None))
```

Comment:
Need to not provide a default argument to cause StopIteration to be raised.

```suggestion
            key, value = next(iter(payload_dict.items()))
```